httomo/wrappers_class.py

```diff
@@ -287,16 +290,37 @@ def __init__(
             if function_name == "rotation":
                 self.wrapper_method = super()._execute_rotation
 
-        # httomolib can include GPU/CuPy methods as as well as the CPU ones. Here
-        # we check if the method can accept CuPy arrays by looking into docstrings.
+        # httomolib exports metadata from the method decorator, which we can use to
+        # check if we support CuPy
+        func = getattr(self.module, method_name)
+        self.meta = func.meta
```

Comment:
what happens if the meta data is not available as with the TomoPy module? Or is it initialised somewhere?

Reply:
This is part of the `HttomolibWrapper`, so this only happens within the httomolib methods, where the `meta` field is available. But this code isn't used at this point - will integrate that today and see it it can all work together.